python-package/lightgbm/callback.py

```diff
@@ -198,8 +199,17 @@ def _init(env):
                 cmp_op.append(lt)
 
     def _callback(env):
+        enabled = not any((boost_alias in env.params
+                           and env.params[boost_alias] == 'dart') for boost_alias in ('boosting',
+                                                                                      'boosting_type',
+                                                                                      'boost'))
```

Comment:
maybe this can be moved into _init(env) ?

Reply:
@guolinke Sorry, I was in a hurry and mislooked [this aspect](https://stackoverflow.com/questions/5218895/python-nested-functions-variable-scoping), which made me think that it's impossible to set `enabled` variable in `_init` function. I've reworked this.